AmazingMeSample/app/src/main/java/amazingme/activities/app/LoginActivity.java

```diff
@@ -0,0 +1,62 @@
+package amazingme.activities.app;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+
+import com.amazingme.activities.R;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+
+import amazingme.app.EnumeratedActivity;
+import amazingme.database.FirebaseHelper;
+import amazingme.model.AmazingMeAppCompatActivity;
+
+/**
+ * Created by yudawinata on 8/31/17.
+ */
+
+public class LoginActivity extends AmazingMeAppCompatActivity {
+    private EditText emailEditText, passwordEditText;
+    private Button loginBtn, registerBtn;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+
+        emailEditText = (EditText) findViewById(R.id.emailEditText);
+        passwordEditText = (EditText) findViewById(R.id.passwordEditText);
+        loginBtn = (Button) findViewById(R.id.loginBtn);
+        registerBtn = (Button) findViewById(R.id.registerBtn);
+
+        registerBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                final Intent registerIntent = new Intent(getApplicationContext(), RegisterActivity.class);
```

Comment:
what's the final for?

Reply:
lol sorry I didn't even notice, this has become sorta a habit of mine. this was a coding standard at Amazon (to make every variable final, when possible), I think the reasoning behind it is to prevent accidental variable re-assignment since most of the time such a thing is unintentional.

I could get rid of them if it's confusing. I don't know how much you care about setting our own coding style.

Reply:
hmmm ok. 

i can dig it. keep it. and keep others on top of that so we don't miss that.